### PR TITLE
Add post page

### PR DESCRIPTION
**Pull request by mashakiselyova**

### CODE REVIEW

File: TestProject.BL/Mappers/RichPostMapper.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        public RichPostModel ToBlModel(Post post)
+        {
+            return new RichPostModel
+            {
+                Id = post.Id,
+                Title = post.Title,
+                Content = post.Content,
+                CreateDate = post.CreateDate,
+                UpdateDate = post.UpdateDate,
+                Author = new Author
```
Comment (dd221x0) [inline TestProject.BL/Mappers/RichPostMapper.cs]: You can create a separate `User` <-> `Author` mapper and use here

File: TestProject/ClientApp/src/components/Layout.js [hunk 2]
```diff
@@ -27,14 +30,30 @@ function Layout() {
             });
     }, [])
 
+    async function handleGetUpdatedRating(postId) {
```
Comment (dd221x0) [inline TestProject/ClientApp/src/components/Layout.js]: `handle` is superfluous here

Comment (dd221x0) [reply TestProject/ClientApp/src/components/Layout.js]: and this shouldn't be placed here, you can create a separate service for such a requests

File: TestProject/Controllers/PostController.cs [hunk 3]
```diff
@@ -83,5 +86,12 @@ public async Task<IActionResult> Delete([FromRoute] int id)
             await _postService.Delete(id);
             return Ok();
         }
+
+        [Route("/posts/getRichPost/{id}")]
+        public RichPostDisplayModel GetRichPost([FromRoute] int id)
+        {
+            return _richPostDisplayMapper.ToWebModel(_postService.GetRichPost(id, 
+                User.Identity.IsAuthenticated ? User.GetEmail() : null));
```
Comment (dd221x0) [inline TestProject/Controllers/PostController.cs]: don't make such a complex expressions, extract variables
📝

File: TestProject/ClientApp/src/components/Posts.js [hunk 4]
```diff
@@ -20,20 +20,13 @@ function Posts({ userProfile, filterByCurrentUser = false }) {
...
-                    const index = newPosts.map(p => p.id).indexOf(postId);
-                    newPosts[index].totalRating = data.totalRating;
-                    newPosts[index].selectedRating = data.ratingByCurrentUser;
-                    setPosts(newPosts);
-                });
-            })
-            .catch(() => {
-                NotificationManager.error("Couldn't get rating");
-            });
+    async function handleUpdateRating(postId) {
```
Comment (dd221x0) [inline TestProject/ClientApp/src/components/Posts.js]: `handle` is not needed here as well

File: TestProject/ClientApp/src/components/PostPage.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import { NotificationContainer, NotificationManager } from 'react-notifications';
+import 'react-notifications/lib/notifications.css';
+import Rating from "./Rating";
+
+function PostPage({ postId, userProfile, getUpdatedRating }) {
+    const [post, setPost] = useState({
+        id: 0,
+        title: "",
+        content: "",
+        createDate: "",
```
Comment (dd221x0) [inline TestProject/ClientApp/src/components/PostPage.js]: it definitely should not be set to empty string

**Review by dd221x0 — CHANGES_REQUESTED** [review]

File: TestProject.BL.Test/Services/PostServiceTests.cs [hunk 6]
```diff
@@ -121,6 +126,39 @@ public async Task Should_delete_post()
...
+            _mockUserRepository.Setup(repo => repo.Get(It.IsAny<Func<User, bool>>()))
+                .Returns((Func<User, bool> predicate) => users.Where(predicate).ToList());
+
+            var result = _postService.GetRichPost(1, "email");
+
+            result.Should().BeEquivalentTo(expected);
+        }
+
+        [Fact]
+        public void If_rich_post_does_not_exist_should_throw_exception()
```
Comment (dd221x0) [inline TestProject.BL.Test/Services/PostServiceTests.cs]: nit: Should_throw_exception_when_requesting_rich_post_that_does_not_exist

File: TestProject/ClientApp/src/components/Layout.js [hunk 7]
```diff
@@ -27,14 +28,28 @@ function Layout() {
             });
     }, [])
 
+    //async function handleGetUpdatedRating(postId) {
```
Comment (dd221x0) [inline TestProject/ClientApp/src/components/Layout.js]: what is this?

Comment (mashakiselyova) [reply TestProject/ClientApp/src/components/Layout.js]: oops

File: TestProject/ClientApp/src/services/RatingService.js [hunk 8]
```diff
@@ -0,0 +1,5 @@
+﻿export default async function getRating(postId) {
```
Comment (dd221x0) [inline TestProject/ClientApp/src/services/RatingService.js]: why is this default export?

File: TestProject/ClientApp/src/components/PostPage.js [hunk 9]
```diff
@@ -2,19 +2,10 @@
...
-        content: "",
-        createDate: "",
-        updateDate: "",
-        updateRating: "",
-        author: { firstName: "", lastName: "" },
-        selectedRating: 0,
-        totalRating: 0
-    });
+function PostPage({ postId, userProfile }) {
+    const [post, setPost] = useState(undefined);
```
Comment (dd221x0) [inline TestProject/ClientApp/src/components/PostPage.js]: You never need to set `undefined`, it is default value. It's better use an empty object in this case.

**Pull request merged**